src/components/TodoForm.tsx

```diff
@@ -0,0 +1,76 @@
+import React, {
+  FormEvent, useEffect, useState,
+} from 'react';
+import { addTodo } from '../api/todos';
+import { Todo } from '../types/Todo';
+import { TContext, useTodoContext } from '../context/TodoContext';
+
+export const TodoForm: React.FC = () => {
+  const [title, setTitle] = useState<string>('');
+  const [isSubmitting, setIsSubmitting] = useState<boolean>(false);
+  const {
+    todos,
+    setTodos,
+    handleError,
+    setTempTodos,
+    titleInputRef,
+  } = useTodoContext() as TContext;
+
+  let counter: number;
+
+  useEffect(() => {
+    if (!isSubmitting) {
+      titleInputRef.current?.focus();
+    }
+  }, [isSubmitting]);
+
+  const handleSubmit = (event: FormEvent<HTMLFormElement>) => {
+    event.preventDefault();
+
+    const newTodo: Todo = {
+      userId: 11550,
+      id: 123 + counter,
+      title: title.trim(),
+      completed: false,
+    };
+
+    if (title.trim().length === 0) {
+      handleError('Title should not be empty');
+    } else {
+      setTempTodos({ ...newTodo, id: 0 });
+      setIsSubmitting(true);
+
+      addTodo(newTodo)
+        .then((res) => {
+          setTodos([...todos, res]);
+          setTitle('');
+        })
+        .catch((error) => {
+          handleError('Unable to add a todo');
+          // eslint-disable-next-line no-console
+          console.log(error);
+        })
+        .finally(() => {
+          setIsSubmitting(false);
+          setTempTodos(null);
+          counter += 1;
```

Comment:
In overall this counter has no effect because is always null. If you want to get next ID then you should check it against the last item in the todos collection.

Reply:
Actually as I can see you don't have to pass ID to the `POST`

Please refer to this documentation link: https://mate-academy.github.io/fe-students-api/todos.html